src/Disks/HDFS/ReadIndirectBufferFromHDFS.h

```diff
@@ -0,0 +1,142 @@
+#pragma once
+
+#include <IO/ReadBufferFromFile.h>
+#include <Disks/IDiskRemote.h>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int CANNOT_SEEK_THROUGH_FILE;
+}
+
+
+/// Reads data from HDFS using stored paths in metadata.
+class ReadIndirectBufferFromHDFS final : public ReadBufferFromFileBase
```

Comment:
Looks like a copy-paste from `DiskS3::ReadIndirectBuffer`.
Can we extract a common class like `ReadIndirectBufferFromRemoteFS` and pass e.g. lambda that will be used for creating underlying (S3/HDFS) ReadBuffer?